numpy/core/overrides.py

```diff
@@ -58,16 +58,6 @@ def get_overloaded_types_and_args(relevant_args):
                 overloaded_types = [arg_type]
                 overloaded_args = [arg]
 
-    # Short-cut for the common case of only ndarray.
```

Comment:
Just to be sure I understand: this (minor) optimization is enabled by your `ndarray.__array_function__` refactor, but isn't required?

Reply:
Indeed, I moved it to the actual execution part as it didn't make sense to check whether there are any non-ndarray overrides here.